client/src/pages/SignUp/index.tsx

```diff
@@ -25,8 +39,137 @@ import {
   PhoneInputWrapper,
   PhoneInput,
 } from './index.style';
+import { LeftDoodles, RightDoodles } from './index.fc';
+import { postUser } from '~/lib/api/users';
+import { useHistory } from '~/core/Router';
 
 const SignUpPage: FC = () => {
+  const { push } = useHistory();
+  const [id, idWarning, handleId] = useInputValidator('', (idIn) => {
+    if (idIn === '') return ' ';
+    if (REG_ID.test(idIn)) return '';
+
+    return WARNING_ID;
+  });
+  const [pw, pwWarning, handlePW] = useInputValidator('', (pwIn) => {
+    if (pwIn === '') return ' ';
+    if (REG_PW.test(pwIn)) return '';
+
+    return WARNING_PW;
+  });
+
+  const [pwRe, pwReWarning, handlePWRe] = useInputValidator(
+    '',
+    (pwReIn) => {
+      if (pwReIn === pw) return '';
+      return WARNING_PWRE;
+    },
+    [pw],
+  );
+
+  const [email, emailWarning, handleEmail] = useInputValidator('', (mailIn) => {
+    if (mailIn === '') return ' ';
+    if (REG_EMAIL.test(mailIn)) return '';
+
+    return WARNING_EMAIL;
+  });
+
+  const [ph0, ph0Warning, handlePh0] = useInputValidator('', (ph0In) => {
+    if (ph0In === '') return ' ';
+    if (REG_PH0.test(ph0In)) return '';
+
+    return WARNING_PHONE;
+  });
+
+  const [ph1, ph1Warning, handlePh1] = useInputValidator('', (ph1In) => {
+    if (ph1In === '') return ' ';
+    if (REG_PH1.test(ph1In)) return '';
+
+    return WARNING_PHONE;
+  });
+
+  const [ph2, ph2Warning, handlePh2] = useInputValidator('', (ph2In) => {
+    if (ph2In === '') return ' ';
+    if (REG_PH2.test(ph2In)) return '';
+
+    return WARNING_PHONE;
+  });
+
+  const [checkAll, setCheckAll] = useState(false);
+  const [check1, setCheck1] = useState(false);
+  const [check2, setCheck2] = useState(false);
```

Comment:
폼으로 묶어서 사용하는 것과 각 상태로 두는 것에 어떤 차이가 있을까요??
성능적으로 차이가 있을라나...? 그냥 궁금해서 남깁니다

```suggestion
  const [check, setCheck] = useState({
    'all' : false,
    'first': false,
    'second': false
  })
```

Reply:
클래스형 state 방식이군요. 
setCheck를 사용하는 방식이 묶어서했을때 불편해서 저는 나눠두는걸 좋아해요.
성능적으로는 똑같을거같네요